analyzer/codechecker_analyzer/analyzers/gcc/analyzer.py

```diff
@@ -0,0 +1,274 @@
+# -------------------------------------------------------------------------
+#
+#  Part of the CodeChecker project, under the Apache License v2.0 with
+#  LLVM Exceptions. See LICENSE for license information.
+#  SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# -------------------------------------------------------------------------
+"""
+"""
+from collections import defaultdict
+# TODO distutils will be removed in python3.12
+from distutils.version import StrictVersion
+import os
+import pickle
+import shlex
+import subprocess
+
+from codechecker_common.logger import get_logger
+
+from codechecker_analyzer import analyzer_context
+
+from .. import analyzer_base
+from ..flag import has_flag
+
+from .config_handler import GccConfigHandler
+from .result_handler import GccResultHandler
+
+LOG = get_logger('analyzer.gcc')
+
+
+class Gcc(analyzer_base.SourceAnalyzer):
+    """
+    Constructs the Gcc analyzer commands.
+    """
+
+    ANALYZER_NAME = 'gcc'
+
+    @classmethod
+    def analyzer_binary(cls):
+        return analyzer_context.get_context() \
+            .analyzer_binaries[cls.ANALYZER_NAME]
+
+    @classmethod
+    def get_version(cls, env=None):
+        """ Get analyzer version information. """
+        return cls.__get_analyzer_version(cls.analyzer_binary(), env)
+
+    def add_checker_config(self, checker_cfg):
+        LOG.error("Checker configuration for Gcc is not implemented yet")
+
+    def get_analyzer_mentioned_files(self, output):
```

Comment:
I think it's not necessary here. This is a ClangSA-specific thing for gathering other .c files that are needed for CTU.